Scripts/Modes/ShootMania/RabbitChase.Script.txt

```diff
@@ -1,325 +1,394 @@
+    if (S_InstagibMode) {
+        SetPlayerAmmo(_Player, CSmMode::EWeapon::Laser, GetPlayerAmmoMax(_Player, CSmMode::EWeapon::Laser));
+    }
+
+    declare Text Message = TextLib::Compose(_("$<%1$> is now the Rabbit"), _Player.User.Name);
+    Message::SendBigMessage(Message, 3000, 2, CUIConfig::EUISound::Notice, 0);
+}
+
+Integer GetRabbitMaxArmor(Integer _PlayerCount) {
+    declare Integer ArmorValue = 0;
+    if (S_DynamicRabbitArmor) {
+        ArmorValue = Math::Clamp(100 * (_PlayerCount / 3), 100, 100 * S_RabbitArmor);
+    } else {
+        ArmorValue = Math::Max(100, 100 * S_RabbitArmor);
+    }
+    return ArmorValue;
+}
+
+Text RabbitStatusML(CSmPlayer _Player) {
+    declare Text RabbitName = "None";
+    if (_Player != Null) {
+        RabbitName = _Player.User.Name;
+    }
+    declare Text MLText = """<label posn="0 -65" halign="center" style="TextStaticVerySmall" text="$sCurrent Rabbit: $ccc$<{{{RabbitName}}}$>" />
+    """;
+    return MLText;
+}
+
+Text RabbitMarkerML() {
+    return """
+    <?xml version="1.0" encoding="utf-8" standalone="yes" ?>
+	<manialink version="3">
+		<frame id="Frame_RabbitMarker" scale="0.6" pos="0, -100">
+			<frame id="Frame_RabbitHead">
+                <label pos="0.5 -5" z-index="0" size="5 5" text="3" halign="center" valign="center" textsize="3" textcolor="000" rot="90" textfont="BiryaniDemiBold"/>
+                <label pos="1 -2.5" z-index="0" size="5 5" text="=" textcolor="F00" rot="90" halign="center" valign="center" textfont="BiryaniDemiBold" textsize="6"/>
+                <label pos="-2 0" z-index="0" size="15 15" text="" halign="center" valign="center" textsize="10" rot="160" id="Label_EarLeft"/>
+                <label pos="2 0" z-index="0" size="15 15" text="" halign="center" valign="center" textsize="10" rot="-160" id="Label_EarRight"/>
+                <label pos="0 -4" z-index="0" size="15 15" text="" halign="center" valign="center" textsize="5" id="Label_Head"/>
+			</frame>
+			<label pos="0 -10" z-index="0" size="30 5" text="UserName" halign="center" valign="center" id="Label_RabbitName"/>
+			<quad pos="0 -10" z-index="0" size="35 5" bgcolor="555555AA" halign="center" valign="center"/>
+		</frame>
+        <script><!--
+        main() {
+            declare CMlLabel Label_RabbitName <=> (Page.GetFirstChild("Label_RabbitName") as CMlLabel);
+            declare CMlLabel Label_EarLeft <=> (Page.GetFirstChild("Label_EarLeft") as CMlLabel);
+            declare CMlLabel Label_EarRight <=> (Page.GetFirstChild("Label_EarRight") as CMlLabel);
+            declare CMlLabel Label_Head <=> (Page.GetFirstChild("Label_Head") as CMlLabel);
+            declare CMlFrame Frame_RabbitMarker <=> (Page.GetFirstChild("Frame_RabbitMarker") as CMlFrame);
+            while (True) {
+                yield;
+                declare netread Text RabbitPlayerLogin for Teams[0];
+                declare Text PreviousLogin;
```

Comment:
Could these variables be declared and initialized outside of the loop so they don't need to get re-declared every frame?
Also, the way it is now seems like it would run the logic to update the label every frame since PreviousLogin is redeclared